### PR TITLE
[ASTGen] Statements

### PR DESCRIPTION
**Pull request by rintaro**

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test. Linux

**Comment by rintaro** [comment]
@swift-ci Please smoke test Linux

File: test/ASTGen/stmts.swift [hunk 1]
```diff
@@ -33,3 +32,112 @@ func test1(e b: Bool) {
...
+  var property: T {
+    _read {
+      yield storedProperty!
+    }
+    _modify {
+      yield &storedProperty!
+    }
+  }
+
+/* FXIME: yield(...) is parsed as an expression. ASTGen needs to onvert it to a statments.
```
Comment (ahoppen) [inline test/ASTGen/stmts.swift:132]: We should fix that in the new parser to parse `yield()` as a yield stmt, I think. Or do we need more context knowledge to decide whether `yield(…)` is a statement than we have in the new parser/

**Review by ahoppen — COMMENTED** [review]
Looks good from my side, just skimmed the Swift parts though.

**Comment by rintaro** [comment]
@swift-ci Please smoke test Linux

File: test/ASTGen/stmts.swift (hunk 1 above)
Comment (rintaro) [reply test/ASTGen/stmts.swift:132]: Yeah, it's contextual. `yield` statements only appear in coroutine accessors (`_read` and `_modify`) https://github.com/apple/swift/blob/a6c98879abefd42d67aeeb3394784281061d1647/include/swift/Parse/Parser.h#L609-L613
I'm on the fence whether `SwiftParser` should recognize this context.

Comment (ahoppen) [reply test/ASTGen/stmts.swift:132]: Oh, that’s … an unfortunate choice in the old parser

**Pull request merged**